pkg/scheduler/core/generic_scheduler.go

```diff
@@ -51,22 +51,71 @@ func (g *genericScheduler) Schedule(ctx context.Context, placement *policyv1alph
 		return result, fmt.Errorf("no clusters available to schedule")
 	}
 
-	feasibleClusters, err := g.findClustersThatFit(ctx, g.scheduleFramework, placement, &spec.Resource, clusterInfoSnapshot)
+	// 1. calculate reserved clusters and candidate clusters
+	// 2. prioritize candidate clusters
+	// 3. select target clusters from (reserved clusters + candidate clusters)
+	// 4. assign replicas
+	readyClusters, unreadyClusters, existingClusters := clusterInfoSnapshot.InspectClusters()
+	bindClusters := util.ConvertToClusterNames(spec.Clusters)
+
+	reservedClusters := bindClusters.Intersection(readyClusters)
+	// Remove reserved clusters that don't fit the placement
+	for clusterName := range reservedClusters {
+		clusterObj := clusterInfoSnapshot.GetCluster(clusterName)
+		resMap := g.scheduleFramework.RunFilterPlugins(ctx, placement, &spec.Resource, clusterObj.Cluster())
+		res := resMap.Merge()
+		if !res.IsSuccess() {
+			klog.V(4).Infof("cluster %q is not fit", clusterName)
+			reservedClusters.Delete(clusterName)
```

Comment:
If a cluster does not match the plugins now, will it be removed from scheduling result?

Reply:
Yes, that's what `ReconcileSchedule` did

Reply:
It may be dangerous. Because the replicas in those clusters may be removed immediately and users does not realize.

Let's think about what `k/k` does as a analogy. If a Pod has been scheduled to a Node, and the Node label changes which does not match the Pod's node claim now, e.g., NodeSelector. The scheduler will not evict this Pod. Actually the eviction of non-matched pods are implemented by `descheduler`.

Should we evict workload in these clusters? I think we should be careful and it is risky.

Reply:
I'm not sure, I just try to keep the same behavior as before